src/compas/geometry/pointcloud.py

```diff
@@ -245,11 +248,9 @@ def from_box(cls, box, n):
         True
 
         """
-        points = box.points
-        x, y, z = zip(*points)
-        xmin, xmax = min(x), max(x)
-        ymin, ymax = min(y), max(y)
-        zmin, zmax = min(z), max(z)
+        xmin, xmax = box.xmin, box.xmax
+        ymin, ymax = box.ymin, box.ymax
+        zmin, zmax = box.zmin, box.zmax
```

Comment:
This change is not related to the str stuff, is it?

Reply:
The change is there because I couldn't actually run tests without fixing the broken functions.